src/app/3d/qgspointcloud3dsymbolwidget.cpp

```diff
@@ -569,3 +570,22 @@ void QgsPointCloud3DSymbolWidget::maximumScreenErrorChanged( double maxScreenErr
   mMaximumScreenError = maxScreenError;
   emitChangedSignal();
 }
+
+void QgsPointCloud3DSymbolWidget::showBoundingBoxesChanged( int checkBoxState )
+{
+  Q_UNUSED( checkBoxState );
+  bool showBoundingBoxes = mShowBoundingBoxesCheckBox->isChecked();
+  if ( showBoundingBoxes == mShowBoundingBoxes )
+    return;
+  mShowBoundingBoxes = showBoundingBoxes;
+  emitChangedSignal();
+}
+
+void QgsPointCloud3DSymbolWidget::setShowBoundingBoxes( bool showBoundingBoxes )
+{
+  if ( showBoundingBoxes == mShowBoundingBoxes )
+    return;
+  mShowBoundingBoxesCheckBox->setChecked( showBoundingBoxes );
+  mShowBoundingBoxes = showBoundingBoxes;
```

Comment:
Why are we keeping also mShowBoundingBoxes in the widget? In this way, the current state is stored in two places - in the widget and in mShowBoundingBoxes.

I think this function could have just `whileBlocking( mShowBoundingBoxesCheckBox )->setChecked( showBoundingBoxes );` in the body. Also, note the use of `whileBlocking()` - it ensures that termporarily no signal will be emitted from the widget.